packages/insomnia-app/app/plugins/context/request.js

```diff
@@ -147,6 +154,82 @@ export function init(
     getAuthentication(): Object {
       return renderedRequest.authentication;
     },
+    getUploadFileName(): string {
+      return renderedRequest.body.fileName || '';
+    },
+    setUploadFileName(fileName: string): boolean {
+      if (!renderedRequest.body || renderedRequest.body.mimeType !== CONTENT_TYPE_FILE) {
+        return false;
+      }
+      renderedRequest.body.fileName = fileName;
+      return true;
+    },
+    getTextForm(): Array<any> {
+      return (renderedRequest.body.params || [])
+        .filter(p => !p.type || p.type === 'text')
+        .map(({ name, value }) => ({ name, value }));
+    },
+    getFileForm(): Array<any> {
+      return (renderedRequest.body.params || [])
+        .filter(p => p.type === 'file')
+        .map(({ name, fileName }) => ({ name, fileName }));
+    },
+    addFormItem(formItem: any): boolean {
```

Comment:
I wonder if this can be typed, instead of `any`?

```suggestion
    addFormItem(formItem: FormItem): boolean {
```